src/Compilers/CSharp/Portable/Symbols/PublicModel/MethodSymbol.cs

```diff
@@ -332,6 +332,18 @@ INamedTypeSymbol IMethodSymbol.AssociatedAnonymousDelegate
 
         DllImportData IMethodSymbol.GetDllImportData() => _underlying.GetDllImportData();
 
+#nullable enable
+        IMethodSymbol? IMethodSymbol.TryGetCorrespondingExtensionImplementationMethod()
+        {
+            if (!_underlying.IsDefinition || !_underlying.GetIsNewExtensionMember())
```

Comment:
curious why the need for the .IsDefinition check.  Thanks!

Reply:
See assertion in `NamedTypeSymbol.TryGetCorrespondingExtensionImplementationMethod` that we use below